pkg/internal/codegen/parse/index.go

```diff
@@ -184,28 +183,38 @@ type resourceTags struct {
 }
 
 // ParseResourceTag parses the tags in a "+resource=" comment into a resourceTags struct
-func parseResourceTag(tag string) resourceTags {
+func parseResourceTag(t *types.Type) (resourceTags, error) {
 	result := resourceTags{}
-	for _, elem := range strings.Split(tag, ",") {
-		kv := strings.Split(elem, "=")
-		if len(kv) != 2 {
-			log.Fatalf("// +kubebuilder:resource: tags must be key value pairs.  Expected "+
-				"keys [path=<subresourcepath>] "+
-				"Got string: [%s]", tag)
-		}
-		value := kv[1]
-		switch kv[0] {
-		//case "rest":
-		//	result.REST = value
-		case "path":
-			result.Resource = value
-		//case "strategy":
-		//	result.Strategy = value
-		case "shortName":
-			result.ShortName = value
+	for _, comment := range t.CommentLines {
+		if strings.Contains(comment, "+kubebuilder:resource") {
+			path := strings.Replace(comment, "+kubebuilder:resource:", "", -1)
+			for _, elem := range strings.Split(path, ",") {
```

Comment:
add a utility like `GetAnnotation` or `ParseKV` for this if there's not already one (`ParseAnnotationToMap`?)